brave/src/main/java/brave/secondary_sampling/SecondarySampling.java

```diff
@@ -69,20 +75,33 @@ public Builder propagationFactory(Propagation.Factory propagationFactory) {
     }
 
     /**
-     * Optional: This will override what is passed to {@link HttpTracing.Builder#serverSampler(HttpRequestSampler)}
+     * Optional: This will override what is passed to {@link HttpTracing.Builder#serverSampler(SamplerFunction)}
      *
      * <p>This controls the primary sampling mechanism for HTTP server requests. This control is
      * present here for convenience only, as it can be done externally with {@link
      * HttpTracingCustomizer}.
      */
-    public Builder httpServerSampler(HttpRequestSampler httpServerSampler) {
+    public Builder httpServerSampler(SamplerFunction<HttpRequest> httpServerSampler) {
```

Comment:
Unlike `secondarySampler(...)`, for primary there is one method for http and one for grpc. Are you thinking to consolidate this in the future? To ask another way, would you create something like `ipcServerSampler` and let it take care of both?

Reply:
I think I made a comment exactly to the point in a different file on this change. it is just work to do. we can pull into another issue.